std::unique_ptr of forward-declared types

vstgui/vstgui/lib/vstguifwd.h forward-declares a number of abstract base types and then also invalidly declares a bunch of type aliases of `std::unique_ptr` of those forward-declared types. This is not supported by the standard library as the `std::unique_ptr`'s default deleter requires the size of the wrapped type, which is not available for a forward-declared type. It would appear that something about the organization of system headers has changed in some manner with C++23 enabled in the Xcode 15 toolchain such as to reveal this error, such as in the case of including vstgui.h:

```
In file included from vstgui/vstgui/vstgui.h:8:
In file included from vstgui/vstgui/lib/vstguibase.h:552:
In file included from vstgui/vstgui/lib/vstguidebug.h:30:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/memory:898:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/shared_ptr.h:31:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:63:19: error: invalid application of 'sizeof' to an incomplete type 'VSTGUI::IPlatformGraphicsPath'
    static_assert(sizeof(_Tp) >= 0, "cannot delete an incomplete type");
                  ^~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:297:7: note: in instantiation of member function 'std::default_delete<VSTGUI::IPlatformGraphicsPath>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:263:75: note: in instantiation of member function 'std::unique_ptr<VSTGUI::IPlatformGraphicsPath>::reset' requested here
  _LIBCPP_INLINE_VISIBILITY _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
                                                                          ^
In file included from vstgui/vstgui/plugin-bindings/plugguieditor.cpp:6:
In file included from vstgui/vstgui/plugin-bindings/plugguieditor.h:8:
In file included from vstgui/vstgui/plugin-bindings/../vstgui.h:22:
vstgui/vstgui/lib/cgraphicspath.h:96:41: note: in instantiation of member function 'std::unique_ptr<VSTGUI::IPlatformGraphicsPath>::~unique_ptr' requested here
                                   PlatformGraphicsPathPtr&& path = nullptr);
                                                                    ^
In file included from vstgui/vstgui/plugin-bindings/plugguieditor.cpp:6:
In file included from vstgui/vstgui/plugin-bindings/plugguieditor.h:8:
In file included from vstgui/vstgui/plugin-bindings/../vstgui.h:9:
In file included from vstgui/vstgui/lib/cbitmap.h:7:
vstgui/vstgui/lib/vstguifwd.h:333:7: note: forward declaration of 'VSTGUI::IPlatformGraphicsPath'
class IPlatformGraphicsPath;
      ^
```

Similarly in vstgui/lib/platform/mac/macfactory.h `VSTGUI:: MacFactory` has an instance of `std::unique_ptr<Impl>` where `Impl` is forward-declared in the header:

```In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/algorithm:1747:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__algorithm/inplace_merge.h:28:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:63:19: error: invalid application of 'sizeof' to an incomplete type 'VSTGUI::MacFactory::Impl'
    static_assert(sizeof(_Tp) >= 0, "cannot delete an incomplete type");
                  ^~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:297:7: note: in instantiation of member function 'std::default_delete<VSTGUI::MacFactory::Impl>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:263:75: note: in instantiation of member function 'std::unique_ptr<VSTGUI::MacFactory::Impl>::reset' requested here
  _LIBCPP_INLINE_VISIBILITY _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
                                                                          ^
vstgui/vstgui/lib/platform/mac/macfactory.h:15:7: note: in instantiation of member function 'std::unique_ptr<VSTGUI::MacFactory::Impl>::~unique_ptr' requested here
class MacFactory final : public IPlatformFactory
      ^
vstgui/vstgui/lib/platform/mac/macfactory.h:15:7: note: in implicit destructor for 'VSTGUI::MacFactory' first required here
vstgui/vstgui/lib/platform/mac/macfactory.h:142:9: note: forward declaration of 'VSTGUI::MacFactory::Impl'
        struct Impl;
               ^
```

--- a/vstgui/lib/cdrawcontext.cpp
+++ b/vstgui/lib/cdrawcontext.cpp
@@ -714,7 +714,7 @@ void CDrawContext::fillRadialGradient (CGraphicsPath* path, const CGradient& gra
 CGraphicsPath* CDrawContext::createGraphicsPath ()
 {
 	if (impl->device)
-		return new CGraphicsPath (impl->device->getGraphicsPathFactory ());
+		return new CGraphicsPath (impl->device->getGraphicsPathFactory (), nullptr);
 	return nullptr;
 }
 

--- a/vstgui/lib/cgraphicspath.h
+++ b/vstgui/lib/cgraphicspath.h
@@ -92,8 +92,7 @@ public:
 	CRect getBoundingBox ();
 	//@}
 
-	CGraphicsPath (const PlatformGraphicsPathFactoryPtr& factory,
-				   PlatformGraphicsPathPtr&& path = nullptr);
+	CGraphicsPath (const PlatformGraphicsPathFactoryPtr& factory, PlatformGraphicsPathPtr&& path);
 	~CGraphicsPath () noexcept override;
 
 	const PlatformGraphicsPathPtr& getPlatformPath (PlatformGraphicsPathFillMode fillMode);

--- a/vstgui/lib/cstring.h
+++ b/vstgui/lib/cstring.h
@@ -221,6 +221,9 @@ public:
 	bool operator== (const UTF8StringPtr otherString) const;
 	bool operator!= (const UTF8StringPtr otherString) const;
 	bool operator== (UTF8StringView otherString) const;
+	bool operator!= (UTF8StringView otherString) const;
+	bool operator== (const UTF8String& otherString) const;
+	bool operator!= (const UTF8String& otherString) const;
 	operator const UTF8StringPtr () const;
 //-----------------------------------------------------------------------------
 private:
@@ -461,6 +464,26 @@ inline bool UTF8StringView::operator== (UTF8StringView otherString) const
 	if (byteCount && otherString.byteCount && *byteCount != *otherString.byteCount)
 		return false;
 	return operator==(otherString.str);
+}
+
+//------------------------------------------------------------------------
+inline bool UTF8StringView::operator!= (UTF8StringView otherString) const
+{
+	if (byteCount && otherString.byteCount && *byteCount != *otherString.byteCount)
+		return true;
+	return operator!= (otherString.str);
+}
+
+//------------------------------------------------------------------------
+inline bool UTF8StringView::operator== (const UTF8String& otherString) const
+{
+	return otherString == str;
+}
+
+//------------------------------------------------------------------------
+inline bool UTF8StringView::operator!= (const UTF8String& otherString) const
+{
+	return otherString != str;
 }
 
 //-----------------------------------------------------------------------------

--- a/vstgui/lib/platform/mac/cgbitmap.cpp
+++ b/vstgui/lib/platform/mac/cgbitmap.cpp
@@ -270,7 +270,8 @@ CGImageRef CGBitmap::getCGImage ()
 		size_t rowBytes = getBytesPerRow ();
 		size_t bitDepth = 32;
 
-		CGBitmapInfo bitmapInfo = kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Big;
+		CGBitmapInfo bitmapInfo =
+			static_cast<CGBitmapInfo> (kCGImageAlphaPremultipliedFirst) | kCGBitmapByteOrder32Big;
 		image = CGImageCreate (static_cast<size_t> (size.x), static_cast<size_t> (size.y), 8, bitDepth, rowBytes, GetCGColorSpace (), bitmapInfo, bitsDataProvider, nullptr, false, kCGRenderingIntentDefault);
 		dirty = false;
 	}
@@ -300,7 +301,8 @@ CGContextRef CGBitmap::createCGContext ()
 	}
 	if (bits)
 	{
-		CGBitmapInfo bitmapInfo = kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Big;
+		CGBitmapInfo bitmapInfo =
+			static_cast<CGBitmapInfo> (kCGImageAlphaPremultipliedFirst) | kCGBitmapByteOrder32Big;
 		context = CGBitmapContextCreate (bits,
 						static_cast<size_t> (size.x),
 						static_cast<size_t> (size.y),

--- a/vstgui/lib/platform/mac/macfactory.h
+++ b/vstgui/lib/platform/mac/macfactory.h
@@ -16,6 +16,7 @@ class MacFactory final : public IPlatformFactory
 {
 public:
 	MacFactory (CFBundleRef bundle);
+	~MacFactory () noexcept override;
 
 	CFBundleRef getBundle () const noexcept;
 

--- a/vstgui/lib/platform/mac/macfactory.mm
+++ b/vstgui/lib/platform/mac/macfactory.mm
@@ -46,6 +46,8 @@ MacFactory::MacFactory (CFBundleRef bundle)
 	mach_timebase_info (&impl->timebaseInfo);
 }
 
+MacFactory::~MacFactory () noexcept = default;
+
 //-----------------------------------------------------------------------------
 CFBundleRef MacFactory::getBundle () const noexcept
 {

